akka-docs/src/main/paradox/additional/rolling-updates.md

```diff
@@ -0,0 +1,90 @@
+parallel during the update, for example in blue green deployments.
+
+For rolling updates related to Akka dependency version upgrades and the migration guides, please see
+@ref:[Rolling Updates and Akka versions](../project/rolling-update.md)
+
+#### This document covers: 
+* [Serialization Compatibility](#serialization-compatibility)
+* [Cluster Sharding](#cluster-sharding)
+* [Cluster Singleton](#cluster-singleton)
+* [Migrating Untyped to Typed](#migrating-untyped-to-typed)
+* [Cluster Shutdown](#cluster-shutdown)
+* [Cluster Configuration Compatibility Check](#cluster-configuration-compatibility-check)
+ 
+## Serialization Compatibility
+
+There are two parts of Akka that need careful consideration when performing an rolling update.
+
+1. Compatibility of remote message protocols. Old nodes may send messages to new nodes and vice versa.
+1. Serialization format of persisted events and snapshots. New nodes must be able to read old data, and
+   during the update old nodes must be able to read data stored by new nodes.
+
+There are many more application specific aspects for serialization changes during rolling upgrades to consider. 
+For example, whether to allow dropped messages or tear down the TCP connection when the manifest is unknown.
+
+* When some message loss during a rolling upgrade is acceptable versus a full shutdown and restart, assuming the application recovers afterwards 
+    - If a `java.io.NotSerializableException` is thrown in `fromBinary` this is treated as a transient problem, the issue logged and the message is dropped
+    - If other exceptions are thrown it can be an indication of corrupt bytes from the underlying transport, and the connection is broken
+* For more zero-impact rolling upgrades, it is important to consider a strategy for serialization format that can be evolved. You can find advice in
+@ref:[Persistence - Schema Evolution](../persistence-schema-evolution.md), which also applies to
+remote messages when deploying with rolling updates.
+
+One approach to retiring a serializer without downtime is carried out in @ref:[two rolling upgrade steps to switch to the new serializer](../serialization.md#rolling-upgrades). 
+
+## Cluster Sharding
+
+During a rolling upgrade, sharded entities receiving traffic may be moved during @ref:[shard rebalancing](../cluster-sharding.md#shard-rebalancing), 
+to an old or new node in the cluster, based on the pluggable allocation strategy and settings.
+When an old node is stopped the shards that were running on it may be allocated to one of the
+other old nodes remaining in the cluster. See @ref[ClusterSingleton](#cluster-singleton) for a useful `ShardCoordinator` optimization.
+
+There are some cases when @ref:[a full cluster restart is needed](../cluster-sharding.md#rolling-upgrades).
+
+## Cluster Singleton
+
+It's more efficient to avoid moving a `ClusterSingleton` more than necessary because they typically have to recover their state
```

Comment:
Rewording suggestion for the whole section:

> Cluster singletons are always running on the oldest node. To avoid moving cluster singletons more than necessary during a rolling upgrade, it is recommended to upgrade the oldest node last. This way cluster singletons are only moved once during a full rolling upgrade. Otherwise, in the worst case cluster singletons may be migrated from node to node which requires coordination and initialization overhead several times.